### PR TITLE
fix: don't detach completion popup while typing when prefix is empty

### PR DESCRIPTION
**Pull request by akoreman**
*Issue #, if available:* NA

*Description of changes:* Currently, when auto showing completion during typing, the popup gets detached when the prefix is `""`. This can be problematic in situations when completions are coming in while typing sentences (e.g. when using AI code generation). When you pause after typing a space and a completion comes in from the completer, the popup will be detached because the prefix at that point in time is `""`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Pull Request Checklist:
* [x] No backwards incompatible changes were made to Ace's public interface which is defined through the main typings file (`ace.d.ts`) and its references:
    * https://github.com/ajaxorg/ace/blob/master/ace.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-modes.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-extensions.d.ts

### CODE REVIEW

File: src/autocomplete.js [hunk 1]
```diff
@@ -362,8 +362,6 @@ class Autocomplete {
             if (this.tooltipNode) {
                 this.updateDocTooltip();
             }
-        } else if (keepPopupPosition && !prefix) {
-            this.detach();
```
Comment (andredcoliveira) [inline src/autocomplete.js:366]: Looks like this was added in https://github.com/ajaxorg/ace/commit/bd95f782d8fd5be28334e8783bba52406d95654c to close the popup when backspacing into an empty string. E.g.:

```
// type something like `con`
con
// press CTRL+Space -> pop-up renders with suggestions like "console.(...)"
// backspace 3 times
<empty line>
// what happens to the popup here after this PR?
```

Comment (akoreman) [reply src/autocomplete.js:366]: yeah it does change the behavior there

https://github.com/ajaxorg/ace/assets/34573235/b7d049df-2e1a-4691-931d-2feddd821a0c

Comment (akoreman) [reply src/autocomplete.js:366]: In some sense, backspacing until it becomes an empty prefix is similar to typing until it's an empty prefix, might be hard to keep the old behavior while backspacing while still allowing for empty prefix completion while typing ahead.

Comment (andredcoliveira) [reply src/autocomplete.js:366]: Maybe we could do something similar to the logic in https://github.com/ajaxorg/ace/blob/6e870fd08e06be47f9da971743d13b25c41f99c2/src/ext/language_tools.js#L140-L153?

I.e., different behaviour depending on whether it's a deletion or an insertion of text

Comment (andredcoliveira) [reply src/autocomplete.js:366]: Maybe a command for `"Backspace"` close to the other commands: https://github.com/ajaxorg/ace/blob/6e870fd08e06be47f9da971743d13b25c41f99c2/src/autocomplete.js#L738-L757

Something along the lines of "if backspace and no prefix then detach"

Comment (akoreman) [reply src/autocomplete.js:366]: I went for the second option in a new commit

**Review by andredcoliveira — APPROVED** [review]
Looks good to me :)

**Review by oykuyilmaz — APPROVED** [review]
